be/src/vec/functions/function_truncate.h

```diff
@@ -1,245 +0,0 @@
-// regarding copyright ownership.  The ASF licenses this file
-// to you under the Apache License, Version 2.0 (the
-// "License"); you may not use this file except in compliance
-// with the License.  You may obtain a copy of the License at
-//
-//   http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing,
-// software distributed under the License is distributed on an
-// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
-// KIND, either express or implied.  See the License for the
-// specific language governing permissions and limitations
-// under the License.
-
-#include <cstddef>
-#include <functional>
-#include <type_traits>
-#include <utility>
-
-#include "common/exception.h"
-#include "common/status.h"
-#include "olap/olap_common.h"
-#include "round.h"
-#include "vec/columns/column.h"
-#include "vec/columns/column_const.h"
-#include "vec/columns/column_decimal.h"
-#include "vec/columns/column_vector.h"
-#include "vec/common/assert_cast.h"
-#include "vec/core/call_on_type_index.h"
-#include "vec/core/field.h"
-#include "vec/core/types.h"
-#include "vec/data_types/data_type.h"
-#include "vec/data_types/data_type_decimal.h"
-#include "vec/data_types/data_type_number.h"
-
-namespace doris::vectorized {
-
-struct TruncateFloatOneArgImpl {
-    static constexpr auto name = "truncate";
-    static DataTypes get_variadic_argument_types() { return {std::make_shared<DataTypeFloat64>()}; }
-};
-
-struct TruncateFloatTwoArgImpl {
-    static constexpr auto name = "truncate";
-    static DataTypes get_variadic_argument_types() {
-        return {std::make_shared<DataTypeFloat64>(), std::make_shared<DataTypeInt32>()};
-    }
-};
-
-struct TruncateDecimalOneArgImpl {
-    static constexpr auto name = "truncate";
-    static DataTypes get_variadic_argument_types() {
-        // All Decimal types are named Decimal, and real scale will be passed as type argument for execute function
-        // So we can just register Decimal32 here
-        return {std::make_shared<DataTypeDecimal<Decimal32>>(9, 0)};
-    }
-};
-
-struct TruncateDecimalTwoArgImpl {
-    static constexpr auto name = "truncate";
-    static DataTypes get_variadic_argument_types() {
-        return {std::make_shared<DataTypeDecimal<Decimal32>>(9, 0),
-                std::make_shared<DataTypeInt32>()};
-    }
-};
-
-template <typename Impl>
-class FunctionTruncate : public FunctionRounding<Impl, RoundingMode::Trunc, TieBreakingMode::Auto> {
-public:
-    static FunctionPtr create() { return std::make_shared<FunctionTruncate>(); }
-
-    ColumnNumbers get_arguments_that_are_always_constant() const override { return {}; }
-    // SELECT number, truncate(123.345, 1) FROM number("numbers"="10")
-    // should NOT behave like two column arguments, so we can not use const column default implementation
-    bool use_default_implementation_for_constants() const override { return false; }
```

Comment:
`use_default_implementation_for_constants` should be override, and return value should be `false`. The reason is clear in comment.